### ISSUE TITLE
DJango API Documentation: Initial Setup (peter)

### ISSUE THREAD

**Issue by jedpittman** [codeforboston/jobhopper#27]
The API: reads the db and serves back JSON.

Configure and research api provider within django. 
build api for a test schema
setup against test schema
document process 

--- 
Review the documentation provided in the Readme and ensure it works on your local machine. 
Provide any updates as needed as a pull request to the DEV branch. If you get stuck, just provide that information in the comments.

**Comment by pgibb96** [comment]
So far, I've set up postgres to run, and added a model for the Occupation Transitions data, and imported that data into the table. I've been trying to figure out the api that is currently on the master branch (I believe what Sean screen shared last week, but I haven't had time to go through it with him to understand what he did). We are going to zoom so he can fill me in, and then I'll try to get it working for the Occupation Transition data to practice CRUD operations.

**Comment by nathan-duddles** [comment]
Should I be setting up a local postgres database on my machine named 'jobhopperdatabase'? I've followed up to Step 8 in the Readme and am getting a 'database "jobhopperdatabase" does not exist' from manage.py runserver. Thanks!

**Comment by jedpittman** [comment]
Peter, I think that you've done everything specified in the bullets. Can you document the process to do everything you've done (with a pull request for any changes to the instructions needed?) and then move this over to in-review so someone can take a look (either me or nathan). Once the review looks good, we can comment here that it has been reviewed. 

There are other issues and if you go further down your path, you might overlap with other folks' work and I just want to avoid wasting other folks' time. If the work you want to do is not in the issues list, please feel free to add an issue (or multiple) so we can groom (discuss) them together Tuesday.

Once you get your work to in-review, you should feel free to grab another issue from near the top of the "to do" in the sprint board that you think you can tackle. Thanks

**Comment by pgibb96** [comment]
> Should I be setting up a local postgres database on my machine named 'jobhopperdatabase'? I've followed up to Step 8 in the Readme and am getting a 'database "jobhopperdatabase" does not exist' from manage.py runserver. Thanks!

Yes, so first you need to set up your postgres server. We originally had it running in sqlite which was a bit easier to work with, but the current version requires that you have postgres set up. You can install it from here https://www.postgresql.org/download/ and you can use something like postico (Sean mentioned it last zoom) as a GUI rather than having to run commands through the terminal. You just have to make sure that the name and port of the database you create on your local postgres server matches the ones in the DATABASES section of the jobhopper/jobhopper/settings.py file. It's towards the bottom of the page. Then, you will have connected them, but there won't be any tables. Once you run the migration commands, it should create tables matching the models created in the jobhopper/jobs/models.py.

**Comment by pgibb96** [comment]
> Peter, I think that you've done everything specified in the bullets. Can you document the process to do everything you've done (with a pull request for any changes to the instructions needed?) and then move this over to in-review so someone can take a look (either me or nathan). Once the review looks good, we can comment here that it has been reviewed.
> 
> There are other issues and if you go further down your path, you might overlap with other folks' work and I just want to avoid wasting other folks' time. If the work you want to do is not in the issues list, please feel free to add an issue (or multiple) so we can groom (discuss) them together Tuesday.
> 
> Once you get your work to in-review, you should feel free to grab another issue from near the top of the "to do" in the sprint board that you think you can tackle. Thanks

@jedpittman Sure! I'm at my parents this weekend, but I can do a pull request tomorrow when I can sit down and make sure everything is good to go. So I'll do a pull request for changes to the README, and do you want the updated models.py with the model for the Occupation Transition data, or should I wait on that?

**Comment by jedpittman** [comment]
I'd wait on that piece because the data folks need to decide the layouts etc. And I caught up to you guys but let's make sure that the migration commands make sense because when I created an empty db and updated settings, I found that it didn't want to do any migrations for jobs.

Also, it looks like some windows specific instructions on VENV would help. e.g. for my machine:

C:\Users\jedpi\OneDrive\Documents\GitHub\jobhopper-forked\venv\Scripts>activate 
(venv) C:\Users\jedpi\OneDrive\Documents\GitHub\jobhopper-forked\venv\Scripts>deactivate

So something that helps people understand that would be useful; the onboarding documentation should not require that the person understand python very much, if at all.

**Comment by pgibb96** [comment]
@jedpittman Did it give an error message when you ran the migrations? If so, what did it say? I just made a slight alteration to the models.py, and when I ran the the three commands in the migrations section, this is what I got. 

`(projects) (base) petergibb@macbook-pro jobhopper % python manage.py migrate

Operations to perform:
  Apply all migrations: admin, auth, contenttypes, jobs, sessions
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
(projects) (base) petergibb@macbook-pro jobhopper % python manage.py makemigrations jobs
Migrations for 'jobs':
  jobs/migrations/0005_auto_20200914_1220.py
    - Alter field occleaveshare on occupationtransitions
    - Alter field pi on occupationtransitions
(projects) (base) petergibb@macbook-pro jobhopper % python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, jobs, sessions
Running migrations:
  Applying jobs.0005_auto_20200914_1220... OK`

Also, for VENV, do you only want to include the windows specific activation/deactivation commands you listed?

**Comment by jedpittman** [comment]
Those commands were the only ones that were windows specific. Ill try to to run the process again before Tuesday Peter and if I am still stuck I’ll send you the message so we can get this buttoned up by end of Tuesday night.

**Comment by pgibb96** [comment]
Did Endy's pull request complete this issue? He added the final step which was to go the api/leads url in order to interact with the API. Is there more to this issue? I'll grab another issue for now

**Comment by jedpittman** [comment]
I think this is complete for now. I have been able to follow the readme and I see the leads api working. I'm closing this.
